### PR TITLE
types: remove extra validation in VerifyCommit

### PR DESCRIPTION
**Pull request by melekes**
plus make sure LastCommit is always non-nil

### CODE REVIEW

File: types/validator_set.go [hunk 1]
```diff
@@ -632,8 +632,14 @@ func (vals *ValidatorSet) VerifyCommit(chainID string, blockID BlockID,
 	if vals.Size() != len(commit.Signatures) {
 		return NewErrInvalidCommitSignatures(vals.Size(), len(commit.Signatures))
 	}
-	if err := verifyCommitBasic(commit, height, blockID); err != nil {
-		return err
+
+	// Validate Height and BlockID.
+	if height != commit.Height {
+		return NewErrInvalidCommitHeight(height, commit.Height)
```
Comment (erikgrinaker) [inline types/validator_set.go:638]: Why `New` here?

Comment (melekes) [reply types/validator_set.go:638]: you mean why create a function instead of constructing a struct here?

Comment (erikgrinaker) [reply types/validator_set.go:638]: No, why is the error prefixed with `New`?

Comment (melekes) [reply types/validator_set.go:638]: it's a function, not an error itself

Comment (erikgrinaker) [reply types/validator_set.go:638]: Oh, of course 🤦‍♂️

**Pull request merged**